pywbem_mock/_namespaceprovider.py

```diff
@@ -0,0 +1,412 @@
+        try:
+            if new_instance['CreationClassName'] != new_instance.classname \
+                or new_instance['ObjectManagerName'] != OBJECTMANAGERNAME \
+                or new_instance['ObjectManagerCreationClassName'] != \
+                OBJECTMANAGERCREATIONCLASSNAME \
+                or new_instance['SystemName'] != SYSTEMNAME \
+                or new_instance['SystemCreationClassName'] != \
+                    SYSTEMCREATIONCLASSNAME:
+
+                raise CIMError(
+                    CIM_ERR_INVALID_PARAMETER,
+                    _format("Namespace creation via CreateInstance: "
+                            "Invalid property for CIM_Instance {0!A}",
+                            new_instance))
+
+        except KeyError as ke:
+            raise CIMError(
+                CIM_ERR_INVALID_PARAMETER,
+                _format("Namespace creation via CreateInstance: "
+                        "Missing property for CIM_Instance {0!A}. Exception "
+                        "{1!A}", new_instance, ke))
+
+        ns_dict = NocaseDict({ns: ns for ns in self.cimrepository.namespaces})
+        if new_namespace not in ns_dict:
+            self.cimrepository.add_namespace(new_namespace)
+
+        # Pass NewInstance to the default CreateInstance
+        # Default CreateInstance creates path, validates all key properties
+        # in instance and inserts instance into the repository.
+        return super(CIMNamespaceProvider, self).CreateInstance(namespace,
+                                                                new_instance)
+
+    def ModifyInstance(self, ModifiedInstance,
+                       IncludeQualifiers=None, PropertyList=None):
+        """
+        Modification of CIM_Namespace instance not allowed
+        """
+        raise CIMError(
+            CIM_ERR_NOT_SUPPORTED,
+            _format("Modification of CIM_Namespace {0!A} not allowed. ",
+                    ModifiedInstance.classname))
+
+    def DeleteInstance(self, InstanceName):
+        """
+        Delete the namespace defined by the `Name` keybinding in the
+        InstanceName parameter and remove the instance defined by InstanceName.
+
+        The namespace must be empty to be removed otherwise an exception is
+        generated.
+
+        This method does not allow removing the interop namespace.
+
+        Parameters:
+
+          InstanceName: (:class:`~pywbem.CIMInstanceName`):
+            The instance path of the instance to be deleted with the
+            following attributes:
+
+            * `classname`: Name of the creation class of the instance.
+            * `keybindings`: Keybindings of the instance. The keybinding
+              `Name` must exist. It defines the namespace to be removed.
+            * `namespace`: Name of the CIM namespace containing the instance.
+              Must not be None.
+            * `host`: value ignored.
+
+        Raises:
+            CIMError: CIM_ERR_INVALID_NAMESPACE if the
```

Comment:
the end of the sentence is missing in the description for CIM_ERR_INVALID_NAMESPACE. Is this the case when the interop namespace is attempted to be deleted?

Reply:
DONE

Reply:
KARL FIX all CIMError: etc. as another PR.